docs/_includes/helpers/embed.html

```diff
@@ -0,0 +1,36 @@
+{% assign include_path = include.embed | prepend: '_embeds/' | append: '.md' %}
+{% for embed in site.embeds %}
+  {% if embed.path == include_path %}
+    {% assign embed_code = embed.content %}
+    {% assign embed_title = embed.title %}
+  {% endif %}
+{% endfor %}
+
+<iframe
+  title="{{ embed_title | default: 'Component Preview' }}"
+  src="{{ site.baseurl }}/embeds/{{ include.embed }}"
+  width="100%"
+  height="400"
+  frameborder="0"
+  class="display-block margin-y-2"
+></iframe>
+<script>
+  (() => {
+    /** @type {HTMLIFrameElement} */
+    const iframe = document.currentScript.previousElementSibling;
+    window.addEventListener('message', (event) => {
+      if (event.origin !== window.location.origin || event.source !== iframe.contentWindow) {
+        return;
+      }
+
+      const height = Number(event.data);
+      iframe.height = height;
```

Comment:
do we want to check that `event` has a `data`? or should we send an Object `{ height }` as event data do we could guard with `'height' in event.data` ?

Reply:
It would probably be wise to do some basic validation, yeah. The condition above at least assures that the message originates from the same domain and we can assume for the moment this is the only code sending messages, but including some validation might help avoid future issues if someone wanted to use `postMessage` for something else.